server/app/views/components/Icons.java

```diff
@@ -525,13 +540,81 @@ public static SvgTag questionTypeSvg(QuestionType type) {
       default: // fall through
         icon = Icons.UNKNOWN;
     }
-    return svg(icon);
+    return svg(icon).withId("svg-link-" + icon.name().toLowerCase(Locale.ROOT));
+  }
+
+  public static SvgTag questionTypeSvgLink(QuestionType type) {
+    Icons icon;
+    switch (type) {
+      case ADDRESS:
+        icon = Icons.ADDRESS;
+        break;
+      case CHECKBOX:
+        icon = Icons.CHECKBOX;
+        break;
+      case CURRENCY:
+        return svg(Icons.CURRENCY)
+            .attr("fill", "none")
+            .attr("stroke-linecap", "round")
+            .attr("stroke-linejoin", "round")
+            .attr("stroke-width", "2");
+      case DATE:
+        icon = Icons.DATE;
+        break;
+      case DROPDOWN:
+        icon = Icons.DROPDOWN;
+        break;
+      case EMAIL:
+        icon = Icons.EMAIL;
+        break;
+      case FILEUPLOAD:
+        icon = Icons.FILEUPLOAD;
+        break;
+      case ID:
+        icon = Icons.ID;
+        break;
+      case NAME:
+        icon = Icons.NAME;
+        break;
+      case NUMBER:
+        icon = Icons.NUMBER;
+        break;
+      case RADIO_BUTTON:
+        icon = Icons.RADIO_BUTTON;
+        break;
+      case ENUMERATOR:
+        icon = Icons.ENUMERATOR;
+        break;
+      case STATIC:
+        return svg(Icons.ANNOTATION)
+            .attr("fill", "none")
+            .attr("stroke-linecap", "round")
+            .attr("stroke-linejoin", "round")
+            .attr("stroke-width", "2");
+      case TEXT:
+        icon = Icons.TEXT;
+        break;
+      case PHONE:
+        icon = Icons.PHONE;
+        break;
+      default: // fall through
+        icon = Icons.UNKNOWN;
+    }
+    return svgLink(icon);
```

Comment:
```suggestion
    return new SvgTag().with(new UseTag().attr("href", "#svg-link-" + icon.name().toLowerCase(Locale.ROOT)));
```

This should be the only return for this method. Since nothing else uses the methods `svgLink` or `use` we can just inline this and delete those two new methods I added.

Although since we need to build the id in two places having a method that takes the icon and returns the id would be good to have.

```
private String makeSvgLinkId(Icon icon) { 
  return String.format("svg-link-%s", icon.name().toLowerCase(Locale.ROOT));
}
```

Keep in mind you only need the `#` on the `use` not for the `id` attribute on the other method.